src/x/os/limits.go

```diff
@@ -27,3 +27,12 @@ type ProcessLimits struct {
 	VMMaxMapCount int64  // corresponds to /proc/sys/vm/max_map_count
 	VMSwappiness  int64  // corresponds to /proc/sys/vm/swappiness
 }
+
+// RaiseProcessNoFileToNROpenResult captures the result of trying to
+// raise the process num files open limit to the nr_open system value.
+type RaiseProcessNoFileToNROpenResult struct {
+	RaiseRequired   bool
+	NROpenValue     uint64
+	NoFileMaxValue  uint64
+	NoFileCurrValue uint64
```

Comment:
Maybe a comment here saying that this will be the curr value *before* the raise was performed (if it was performed)

Reply:
That's not true though, it will be the curr value (after the raise) or if no raise then it will be the curr value (unadjusted).